MCDataProducts/inc/SimParticle.hh

```diff
@@ -203,22 +203,24 @@ namespace mu2e {
 
     // the following is for excited ions
     IonDetail const& ion()                      const { return _ion; }
-    double           startExcitationEnergy()    const { return _ion.excitationEnergy;}
+    float           startExcitationEnergy()    const { return _ion.excitationEnergy;}
     int              startFloatLevelBaseIndex() const { return _ion.floatLevelBaseIndex;};
 
     // Information at the end of the track.
-    CLHEP::Hep3Vector const& endPosition() const { return _endPosition;}
-    CLHEP::HepLorentzVector const& endMomentum() const { return _endMomentum;}
-    double       endGlobalTime()  const { return _endGlobalTime; }
-    double&      endGlobalTime()        { return _endGlobalTime; }
-    double       endProperTime()  const { return _endProperTime; }
+    CLHEP::Hep3Vector endPosition() const { return Geom::Hep3Vec(_endPosition);}
+    CLHEP::HepLorentzVector endMomentum() const { return Geom::HepLorentzVec(_endMomentum);}
+    XYZVectorD& endPosXYZ() { return _endPosition;}
+    XYZTVectorF& endMomXYZT() { return _endMomentum;}
```

Comment:
I think these two should return const& and should be const functions.  I don't think we need to modifier versions that are present as coded now.

Reply:
I second this.